### PR TITLE
[ci] add script for running builds on external PRs

### PR DESCRIPTION
**Pull request by schallert**
Until we have a more complex label system for running PRs from external
contributors, this can hold us over.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: .buildkite/scripts/pr_build.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+function cleanup() {
+  rm -f "$RESP"
+}
+
+trap cleanup EXIT
+
+curl -H "Authorization: token $GITHUB_PR_TOKEN" -sSf -o "$RESP" "https://api.github.com/repos/m3db/m3/pulls/$1"
+
+function jq_field() {
+  <"$RESP" jq -r "$1"
```
Comment (mway) [inline .buildkite/scripts/pr_build.sh]: we should fail in `usage()` if `jq` isn't in `PATH`

Comment (schallert) [reply .buildkite/scripts/pr_build.sh]: Hmm we kinda get that for free right now:

```
$ PATH=/usr/bin:/bin .buildkite/scripts/pr_build.sh 1759
...
.buildkite/scripts/pr_build.sh: line 40: jq: command not found
...
```

Comment (mway) [reply .buildkite/scripts/pr_build.sh]: _coincidentally_, it doesn't do anything bad before it fails due to `jq` not being installed. 😉 

agreed that in this case it's nbd, but generally we should do checks beforehand (also, a "jq must be installed and in `$PATH` to use this tool" message is less ambiguous than a "command not found" error).

**Comment by robskillington** [comment]
TY for this, I was looking into how we might be able to leverage web hooks to trigger this too (using labels or some such).

**Pull request merged**